### PR TITLE
fix: properly handle multi-line secrets in secret masking

### PR DESCRIPTION
**Pull request by aaronsteers**
## Summary by CodeRabbit

* **Bug Fixes**
  * Improved CI secret masking to recursively find secrets in nested configurations (dicts/lists) and mask multi-line values line-by-line, preventing accidental exposure while preserving non-secret output and CI gating behavior.

* **Tests**
  * Added comprehensive unit tests validating masking across flat, nested, list-based, multi-line and edge-case configurations.

### CODE REVIEW

File: airbyte_cdk/cli/airbyte_cdk/_secrets.py [hunk 1]
```diff
@@ -459,25 +459,60 @@ def _print_ci_secrets_masks(
...
+        if line.strip():  # Skip empty lines
+            print(f"::add-mask::{line!s}")
+
+
+def _print_ci_secret_mask_for_value(value: Any) -> None:
+    """Print GitHub CI mask for a single secret value.
+
+    Call this function for any values identified as secrets, regardless of type.
+    """
+    if isinstance(value, dict):
```
Comment (dbgold17) [inline airbyte_cdk/cli/airbyte_cdk/_secrets.py:478]: If I understand correctly, this method `_print_ci_secret_mask_for_value` only get's called once we deem we've found a secret. So even if the value happens to be a dict, can we just treat it as one big opaque secret blob for the purposes of masking? i.e. do we need to recurse here?

FWIW, this is the first time I'm looking at how we do secrets masking, so I don't have a great understanding yet.

File: airbyte_cdk/cli/airbyte_cdk/_secrets.py [hunk 2]
```diff
@@ -459,25 +459,60 @@ def _print_ci_secrets_masks(
...
+        # For nested dicts, we call recursively on each value
+        for v in value.values():
```
Comment (dbgold17) [inline airbyte_cdk/cli/airbyte_cdk/_secrets.py:480]: as written, wouldn't this not mask the entire dictionary.

My reading is that given a dictionary deemed to be a secret like this:
```
{
"k1": {
  "k2": "v1",
  "k3": "v2"
}
}
```
we'd only mask the terminal values like this:

```
{
"k1": {
  "k2": "***",
  "k3": "***"
}
}
```

File: airbyte_cdk/cli/airbyte_cdk/_secrets.py [hunk 3]
```diff
@@ -459,25 +459,60 @@ def _print_ci_secrets_masks(
         _print_ci_secrets_masks_for_config(config=config_dict)
 
 
+def _print_ci_secret_mask_for_string(secret: str) -> None:
+    """Print GitHub CI mask for a single secret string.
+
+    We expect single-line secrets, but we also handle the case where the secret contains newlines.
+    For multi-line secrets, we must print a secret mask for each line separately.
+    """
+    for line in secret.splitlines():
```
[the remark this answers is not in the document]
Comment (dbgold17) [reply airbyte_cdk/cli/airbyte_cdk/_secrets.py:468]: oh I just saw your response to Jose asking the same question

File: airbyte_cdk/cli/airbyte_cdk/_secrets.py (hunk 2 above)
Comment (aaronsteers) [reply airbyte_cdk/cli/airbyte_cdk/_secrets.py:480]: @dbgold17 - I did consider this but decided to only mask the values if the secrets are nested.

I tried but could not think of any cases where the key of the dictionary would be a secret on its own. (Again, this is scoped to the manner in which we have users specifying config in our connectors.)

The problem with masking they keys of the dictionary: If the parent is "credentials" and the sub-key is "password" and the value of credentials.password is the password, we do want to mask the password, but masking the word "password" would make it censored everywhere in our logs. So an error like "Missing password" would become "Missing ****", which is not what we want.

File: airbyte_cdk/cli/airbyte_cdk/_secrets.py [hunk 4]
```diff
@@ -459,25 +459,60 @@ def _print_ci_secrets_masks(
...
+        if line.strip():  # Skip empty lines
+            print(f"::add-mask::{line!s}")
+
+
+def _print_ci_secret_mask_for_value(value: Any) -> None:
```
[the remark this answers is not in the document]
Comment (aaronsteers) [reply airbyte_cdk/cli/airbyte_cdk/_secrets.py:473]: @jpefaur - Good question. The only reason I created it is because we have to call it twice in the calling function - once for lists and once for dictionaries. And I wanted to be extra careful to keep things DRY for safety and clarity.

**Review by dbgold17 — APPROVED** [review]
discussed online, approving!
